### DISCUSSION TITLE
String Array

### DISCUSSION THREAD

**Discussion by airnh** [pocketbase/pocketbase#2010] [Q&A] [answered]
Hello,

I'm quite new to PB.  How do you store an array of strings for a field? Do you make another collection and use it as a relationship? Select? or JSON and put the array there. It seems there's no direct way to store just simple array(s) of strings.

Please share your advice.
Best

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to store a dynamic/arbitrary array values you have to use either the `json` type field or create a separate collection with fixed fields (_this is preferable if the array items could be more than just a string and have fixed structure_).

If the array values you want to store are from a predefined list, you can define a multiple `select` field with the possible values as select options.

**Comment by airnh** [comment]
Thanks for the input. Just pure curiosity. I wonder why PB doesn't have the array type for strings or numbers (other than select), which could be a bit more intuitive for some usecases.

**Reply by ganigeorgiev** [reply]
Because I don't have a use case for it and the suggested 2 workarounds are sufficient for now.

The scenarios where you'll want to allow users to submit random strings in my opinion are very rare and you might be better creating a collection for that since you can define more than just a string/number and have a queryable structure of your data (like `tags`, `meta`, etc.).

**Reply by airnh** [reply]
I understand. Thanks alot!
